mycroft/util/qml_file_server.py

```diff
@@ -0,0 +1,84 @@
+# NEON AI (TM) SOFTWARE, Software Development Kit & Application Framework
+# All trademark and other rights reserved by their respective owners
+# Copyright 2008-2022 Neongecko.com Inc.
+# Contributors: Daniel McKnight, Guy Daniels, Elon Gasper, Richard Leeds,
+# Regina Bloomstine, Casimiro Ferreira, Andrii Pernatii, Kirill Hrymailo
+# BSD-3 License
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import socketserver
+import http.server
+
+from tempfile import gettempdir
+from os.path import isdir, join, dirname
+from threading import Thread, Event
+
+_HTTP_SERVER = None
+
+
+class QmlFileHandler(http.server.SimpleHTTPRequestHandler):
+    def end_headers(self) -> None:
+        mimetype = self.guess_type(self.path)
+        is_file = not self.path.endswith('/')
+        if is_file and any([mimetype.startswith(prefix) for
+                           prefix in ("text/", "application/octet-stream")]):
+            self.send_header('Content-Type', "text/plain")
+            self.send_header('Content-Disposition', 'inline')
+        super().end_headers()
+
+
+def start_qml_http_server(skills_dir: str, port: int = 8000):
```

Comment:
I wonder if using port 8000 is a wise thing todo as it is being used as alternative http port and many other programs already use it as well. This might run into conflicts a lot.

Perhaps choose another port as default, just like the 18181 for the websocket?

Reply:
good call, many demo apps use 8000 ...

Reply:
Or make it a config variable just like the gui websocket port and if not set, make it gui websocket +1

Reply:
Oh yeah, thanks for catching this. I meant to put this in a config option too